src/Provider/Youtube/Provider.php

```diff
@@ -147,7 +149,17 @@ public function provide($input)
         }
 
         $response = $this->getHttpClient()->send($request, $options);
-
+		
+		//But just incase it doesn't, we can fallback to the old URL.
+		if (strpos($response->getBodyAsString(), 'status=fail') !== false) {
+			$video_info_url = 'http://www.youtube.com/get_video_info?&video_id=' . $input . '&asv=3&el=detailpage&hl=en_US';
+            $request = $this->getHttpClient()->createRequest(
```

Comment:
`CurlClient::createRequest()` was made private to implement PSR-18 in future. Use `CurlClient::createFullRequest()` instead.

```diff
-             $request = $this->getHttpClient()->createRequest(
+            $request = $this->getHttpClient()->createFullRequest(
```